src/test/java/bio/terra/service/filedata/azure/util/BlobIOTestUtility.java

```diff
@@ -0,0 +1,157 @@
+      TokenCredential tokenCredential, String sourceAccountName, String destinationAccountName) {
+    this.sourceContainerName = generateNewTempContainerName();
+    this.destinationContainerName = generateNewTempContainerName();
+    this.sourceBlobContainerClient =
+        createBlobContainerClient(tokenCredential, this.sourceContainerName, sourceAccountName);
+    this.destinationBlobContainerClient =
+        createBlobContainerClient(
+            tokenCredential, this.destinationContainerName, destinationAccountName);
+    this.sourceBlobContainerClient.create();
+    this.destinationBlobContainerClient.create();
+    this.tokenCredential = tokenCredential;
+  }
+
+  private BlobContainerClient createBlobContainerClient(
+      TokenCredential credential, String containerName, String accountName) {
+
+    return new BlobContainerClientBuilder()
+        .credential(credential)
+        .endpoint(String.format(Locale.ROOT, STORAGE_ENDPOINT_PATTERN, accountName, containerName))
+        .buildClient();
+  }
+
+  public String generateSourceContainerUrlWithSasReadAndListPermissions(String accountKey) {
+    String sasToken = generateContainerSasTokenWithReadAndListPermissions(accountKey);
+
+    return String.format(
+        "%s?%s", this.getSourceBlobContainerClient().getBlobContainerUrl(), sasToken);
+  }
+
+  public String generateContainerSasTokenWithReadAndListPermissions(String accountKey) {
+    BlobSasPermission permissions =
+        new BlobSasPermission().setReadPermission(true).setListPermission(true);
+
+    OffsetDateTime expiryTime = OffsetDateTime.now().plusDays(1);
+    SasProtocol sasProtocol = SasProtocol.HTTPS_ONLY;
+
+    // build the token
+    BlobServiceSasSignatureValues sasSignatureValues =
+        new BlobServiceSasSignatureValues(expiryTime, permissions).setProtocol(sasProtocol);
+
+    BlobContainerClient blobContainerClient =
+        new BlobContainerClientBuilder()
+            .credential(
+                new StorageSharedKeyCredential(
+                    this.sourceBlobContainerClient.getAccountName(), accountKey))
+            .endpoint(this.sourceBlobContainerClient.getBlobContainerUrl())
+            .buildClient();
+
+    return blobContainerClient.generateSas(sasSignatureValues);
+  }
+
+  private String generateNewTempContainerName() {
+    return UUID.randomUUID().toString().toLowerCase().replace("-", "").substring(0, 30);
+  }
+
+  public List<String> uploadSourceFiles(int numOfFiles, long length) {
+    return Stream.iterate(0, n -> n + 1)
+        .limit(numOfFiles)
+        .map(
+            i ->
+                uploadSourceFile(
+                    String.format("%s/%s%s", i, SOURCE_BLOB_NAME, UUID.randomUUID().toString()),
+                    length))
+        .collect(Collectors.toList());
+  }
+
+  public String uploadSourceFile(String blobName, long length) {
+    sourceBlobContainerClient.getBlobClient(blobName).upload(createInputStream(length), length);
```

Comment:
You're never closing your streams

Reply:
The streams here, are for test purposes only, and don't open any handles (e.g. files). They are implementation of `InputStream`  with an overridden implementation of `read` that returns random chars until the length is reached, at which point it returns -1 to indicate completion. So as far as I can tell, there is no risk of lingering instances in the heap or OS.